pkg/server/plugin/notifier/k8sbundle/k8sbundle.go

```diff
@@ -369,6 +394,57 @@ func (c configMapClient) Watch(ctx context.Context, config *pluginConfig) (watch
 	return nil, nil
 }
 
+// apiServiceClient encapsulates the Kubenetes API for updating the CA Bundle in an API Service
+type apiServiceClient struct {
+	*aggregator.Clientset
+}
+
+func (c apiServiceClient) Get(ctx context.Context, namespace, name string) (runtime.Object, error) {
+	return c.ApiregistrationV1().APIServices().Get(ctx, name, metav1.GetOptions{})
+}
+
+func (c apiServiceClient) GetList(ctx context.Context, config *pluginConfig) (runtime.Object, error) {
+	return c.ApiregistrationV1().APIServices().List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=true", config.APIServiceLabel),
+	})
+}
+
+func (c apiServiceClient) CreatePatch(ctx context.Context, config *pluginConfig, obj runtime.Object, resp *hostservices.FetchX509IdentityResponse) (runtime.Object, error) {
+	apiService, ok := obj.(*apiregistration.APIService)
+	if !ok {
+		return nil, k8sErr.New("wrong type, expecting APIService")
+	}
+
+	// Check if APIService needs an update
+	if bytes.Equal(apiService.Spec.CABundle, []byte(bundleData(resp.Bundle))) {
+		return nil, status.Errorf(codes.AlreadyExists, "APIService %s is already up to date", apiService.Name)
```

Comment:
NIT: it i not directly related with your changes (because there are some code that i doing the same)  but we are mixing regular errors (L415) with GRPC Status (L420), I think we must use only one kind of errors (and I prefer GRPC status beause it i a GRPC rpc)